### ISSUE TITLE
Add Example For Getting Name of Enum Key

### ISSUE THREAD

**Issue by amarraff** [godotengine/godot-docs#9654]
**Your Godot version:**
v4.2.2.stable.official [15073afe3]

**Issue description:**
The documentation for enums shows how to print all keys and their values, but not how to print the name of a single key. It would be helpful to display the correct syntax for printing the name of a given key, which I found information on at this forum post:
https://forum.godotengine.org/t/taking-an-enum-name-instead-of-its-value/24261/3

**URL to the documentation page (if already existing):**
https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#enums

**Comment by AThousandShips** [comment]
This isn't standard and depends on the implementation of enums, don't think we should add this, especially as enums can have duplicate keys

**Comment by amarraff** [comment]
> This isn't standard and depends on the implementation of enums, don't think we should add this, especially as enums can have duplicate keys, you should probably use your own manual way to access these, like:
> 
> ```gdscript
> enum MyEnum {
>     KEY_1 = 1,
>     ...
> }
> 
> func get_name(key: MyEnum) -> String:
>     match key:
>         KEY_1:
>             return "KEY_1"
>         ...
> ```

Thank you for the example code; perhaps it would make more sense for me to request an improvement to the engine itself via adding a get_key_name func, or something similar? 

In my case, I have an enum called "Characters" and when I print their dialogue in a text box on screen, I print the name of the character enum. Before I came the code in the forum post I linked, it was just printing the index of the enum. My code looks like this:
`print_debug(str(Enums.Characters.keys()[diob.character])`
This shows me the name of the character the dialogue is relevant to when it begins printing, so I know the right character enum was assigned to the dialogue. It works for what I need; I'm unfamiliar with what duplicate keys you mentioned are, is the issue linked to enums not needing an explicit index?

**Comment by AThousandShips** [comment]
It's fine to *do* this, but it's an implementation detail, so I would be a bit hesitant to document it

With duplicate keys you can easily have:
```gdscript
enum MyEnum {
    FOO = 1,
    BAR = 2,
    BAZ = 1,
}
```
That won't work in reverse, it won't work with `match` either but it's part of the limitations that make having an explicit method to look up things difficult, because there isn't a guarantee that `MyEnum.find_key(BAZ)` actually returns `BAZ`, in this case it would likely return `FOO`

**Comment by amarraff** [comment]
> It's fine to _do_ this, but it's an implementation detail, so I would be a bit hesitant to document it
> 
> With duplicate keys you can easily have:
> 
> ```gdscript
> enum MyEnum {
>     FOO = 1,
>     BAR = 2,
>     BAZ = 1,
> }
> ```
> 
> That won't work in reverse, it won't work with `match` either but it's part of the limitations that make having an explicit method to look up things difficult, because there isn't a guarantee that `MyEnum.find_key(BAZ)` actually returns `BAZ`, in this case it would likely return `FOO`

Right, I wasn't even aware that was a possibility. It would make sense to put a warning about duplicates next to the example, if documented. 

If documenting it is a bad idea, that's okay. I just want to emphasize that I'd likely still be stuck if I didn't come across that forum post, so I thought perhaps an official mention of the syntax in the docs could be a good idea (if it's not already mentioned somewhere I missed).

**Comment by AThousandShips** [comment]
It's mentioned that they are implemented as dictionaries, and they have this method

What I'm saying is just that:
* It's an implementation detail
* It's not free of potential issues
* Most of the time you don't need or want to get the name of an enum, so usually a `match` is the way used

**Comment by amarraff** [comment]
> It's mentioned that they are implemented as dictionaries, and they have this method
> 
> 
> 
> What I'm saying is just that:
> 
> * It's an implementation detail
> 
> * It's not free of potential issues
> 
> * Most of the time you don't need or want to get the name of an enum, so usually a `match` is the way used

I understand more now why you're hesitant to place it under the enums section as an example. It's just not information that was obvious to me despite my existing programming knowledge, so it may help others to have that implementation detail shown/explained somewhere in the docs if it isn't there already. It doesn't have to be under the enum section if that doesn't make sense.

I just tried searching the docs for )[, the symbols between the end of a given func and the beginning of an index query, but it didn't find any results.

I took a look at the documentation for arrays and dictionaries, but neither appears to show the syntax for accessing an index after typing a function which returns an array (at least that I'm seeing). 

I think at least having this mentioned somewhere in the docs would be helpful information since the engine/language is capable of it, and there are use cases for it beyond my own. Something along the lines of "you can also access elements from functions that return arrays, like this:" followed by a short example code.
